pkg/epp/scheduling/types/types.go

```diff
@@ -64,12 +63,21 @@ func (pm *PodMetrics) GetMetrics() *backendmetrics.Metrics {
 	return pm.Metrics
 }
 
+func (pm *PodMetrics) SetScore(score float64) {
+	pm.score = score
+}
+
+func (pm *PodMetrics) Score() float64 {
+	return pm.score
+}
+
 type PodMetrics struct {
+	score float64
 	*backendmetrics.Pod
 	*backendmetrics.Metrics
 }
 
-func NewContext(ctx context.Context, req *LLMRequest, pods []*PodMetrics) *Context {
+func NewContext(ctx context.Context, req *LLMRequest, pods []Pod) *Context {
```

Comment:
consider renaming to `SchedulerContext` or something like that. 
when first seeing context it was confusing (I was expected the golang context.Context).